lib/covalence/core/cli_wrappers/popen_wrapper.rb

```diff
@@ -77,6 +110,37 @@ def spawn_subprocess(env, run_cmd,
         return wait_thread.value.exitstatus
       end
 
+      def spawn_subprocess_with_workdir(env, run_cmd,
+                       stdin_io: STDIN,
+                       stdout_io: STDOUT,
+                       stderr_io: STDERR,
+                       ignore_exitcode: false,
+                       path: Dir.pwd,
+                       workdir: Dir.pwd)
+        logger.info "path: #{path} workdir: #{workdir} run_cmd: #{run_cmd}"
+        Signal.trap("INT") {} #Disable parent process from exiting, orphaning the child fork below
```

Comment:
I think `Signal.trap("INT") {}` just trying to trap `Ctrl-c` but agree if you hit `Ctrl-c` before this, should just probably exit the program at the time you trap this. If you hit `Ctrl-c` after this and before `Open3.popen3(...`, I am not sure it will do anything. If you hit `Crtl-c` after `Open3.popen3(...`, this will be handled by the `Signal.trap('INT')` inside `Open3.popen3`. And trapping it and does nothing it's not the best practice.